bin/jsii-release-golang

```diff
@@ -0,0 +1,159 @@
+  dry_run=true
+fi
+
+if [ ! -d ${dir} ]; then
+  error "${dir} either doesnt exist or is not a directory"
+fi
+
+if [ -z "${GITHUB_TOKEN:-}" ]; then
+  error "GITHUB_TOKEN env variable is required"
+fi
+
+go_mod=""
+versions=("")
+
+# make sure each directory has a go.mod file
+for subdir in $(find ${dir} -mindepth 1 -maxdepth 1 -type d)
+do
+  if [ ! -f ${subdir}/go.mod ]; then
+    error "expected to find go.mod in ${subdir}. make sure you specify the root directory of all your modules."
+  fi
+  # pick one mod file to derive repository
+  go_mod=${subdir}/go.mod
+
+  # collect versions if they exist. used both for validation and for default value.
+  if [ -f ${subdir}/version ]; then
+    version=$(cat ${subdir}/version)
+    versions=(${versions[@]} ${version})
+  fi
+
+done
+
+if [ ${go_mod} = "" ]; then
+  error "No go modules found in ${dir}"
+fi
+
+unique_versions=($(echo "${versions[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
+count_unique_version=$(echo "${#unique_versions[@]}")
+
+if [ ${count_unique_version} -gt 1 ]; then
+  error "multiple versions are not supported. please use a single version across all your modules."
+fi
+
+if [ -z "${VERSION:-}" ]; then
+
+  if [ ${count_unique_version} -eq 0 ]; then
+    error "No version files found. Please supply a version using the VERSION env variable."
+  fi
+
+  # must be a single element at this point
+  VERSION=${unique_versions[0]}
+fi
+
+GIT_COMMIT_MESSAGE=${GIT_COMMIT_MESSAGE:-"chore(release): ${VERSION}"}
+
+module_name=$(grep module ${go_mod} | cut -d ' ' -f 2)
+domain=$(echo ${module_name} | cut -d '/' -f 1)
+
+if [ ${domain} != "github.com" ]; then
+  # the way we clone the repo with the token is a GitHub thing.
+  error "'${domain}' domain is not supported. Only GitHub repositories are allowed"
+fi
+
+if [ -z "${GITHUB_REPO:-}" ]; then
+  owner=$(echo ${module_name} | cut -d '/' -f 2)
+  repo=$(echo ${module_name} | cut -d '/' -f 3)
+  GITHUB_REPO=${owner}/${repo}
+fi
+
+target_repo_dir=$(mktemp -d)/repo
+
+echo "Cloning target repository ${GITHUB_REPO}"
+git clone https://${GITHUB_TOKEN}@github.com/${GITHUB_REPO}.git ${target_repo_dir}
+
+pushd ${target_repo_dir}
+git config user.name ${GIT_USER_NAME}
+git config user.email ${GIT_USER_EMAIL}
+
+# checkout or create
+git checkout ${GIT_BRANCH} || git checkout -b ${GIT_BRANCH}
+
+for subdir in $(find $(pwd) -mindepth 1 -maxdepth 1 -type d)
+do
+  if [ -f ${subdir}/go.mod ]; then
+    # remove to account for deleted files or entire modules
+    echo "Removing ${subdir}"
+    rm -r ${subdir}
+  fi
+done
+
+echo "Copying go modules to repository root"
+cp -r ${dir}/* .
+
+git add .
+git commit -m "${GIT_COMMIT_MESSAGE}"
+
+if $dry_run; then
+  echo "Will push to branch: ${GIT_BRANCH}"
+else
+  git push origin ${GIT_BRANCH}
+fi
+
+tag_name=v${VERSION}
+
+git tag -a ${tag_name} -m $VERSION
+
+if $dry_run; then
+  echo "Will create a tag: ${tag_name}"
+else
+  git push origin ${tag_name}
```

Comment:
Do we also need to push to the branch?

Reply:
You mean should we skip pushing to the branch? 

https://github.com/aws/jsii-release/blob/51f669006f11d91a78df2a1ebad749fc0087446a/bin/jsii-release-golang#L143-L147

Reply:
https://github.com/aws/jsii-release/pull/23/commits/e357e7b42a702ec888d036bcdee671a3ecb8ac3a